core/app/models/spree/item_adjustments.rb

```diff
@@ -58,9 +58,7 @@ def update
       @item.included_tax_total = tax.select(&:included?).map(&:update!).compact.sum
       @item.additional_tax_total = tax.reject(&:included?).map(&:update!).compact.sum
 
-      item_cancellation_total = adjustments.select(&:cancellation?).map(&:update!).compact.sum
```

Comment:
I would imagine tests are failing because of the lost call to `update!`. I do think there is room for improvement on how this is done though.